### PR TITLE
[SPARK-23284][SQL] Document the behavior of several ColumnVector's get APIs when accessing null slot

### PR DESCRIPTION
**Pull request by viirya**
## What changes were proposed in this pull request?

For some ColumnVector get APIs such as getDecimal, getBinary, getStruct, getArray, getInterval, getUTF8String, we should clearly document their behaviors when accessing null slot. They should return null in this case. Then we can remove null checks from the places using above APIs.

For the APIs of primitive values like getInt, getInts, etc., this also documents their behaviors when accessing null slots. Their returning values are undefined and can be anything.

## How was this patch tested?

Added tests into `ColumnarBatchSuite`.

### CODE REVIEW

**Comment by viirya** [comment]
Once map support is added later, we should also document `getMap`'s behavior.

**Comment by viirya** [comment]
cc @cloud-fan @ueshin @kiszk

**Comment by ueshin** [comment]
LGTM so far. Let's wait for map type support.

**Comment by cloud-fan** [comment]
LGTM so far, one thing I wanna add is to also document the behavior of accessing null primitive values, e.g. `getInt`. We can say that the return value is undefined and can be anything, if that slot is null. Same to the batched version like `getInts`.

**Comment by cloud-fan** [comment]
BTW we should also update `ColumnarBatchSuite` for the "return null" behavior

**Comment by viirya** [comment]
Since the map support is added, I'll do related change later.

**Comment by ueshin** [comment]
We also need to add `if (isNullAt(rowId)) return null;` to `WritableColumnVector.getMap()`?

**Comment by viirya** [comment]
@ueshin Yes, missing it. Thanks. I'll add it.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala [hunk 1]
```diff
@@ -1261,4 +1261,140 @@ class ColumnarBatchSuite extends SparkFunSuite {
     batch.close()
     allocator.close()
   }
+
+  testVector("getUTF8String should return null for null slot", 4, StringType) {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala]: we already have test cases for each type, can we just change the existing test cases a little to add this null check?

Comment (viirya) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala]: That sounds ok to me. I'll commit the change by tonight.

Comment (viirya) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala]: Seems we don't have individual tests for binary type and decimal type?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala [hunk 2]
```diff
@@ -1261,4 +1269,38 @@ class ColumnarBatchSuite extends SparkFunSuite {
     batch.close()
     allocator.close()
   }
+
+  testVector("getDecimal should return null for null slot", 4, DecimalType.IntDecimal) {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala]: shall we make it a normal test case for decimal type? we can follow the other tests, e.g. create a decimal array, and check the value of column vector at the same index.

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
ok to test

**Comment by ueshin** [comment]
LGTM.

**Comment by cloud-fan** [comment]
thanks, merging to master/2.3!

**Pull request closed without merging**